### PR TITLE
Reenable --skip-validation.

### PR DESCRIPTION
**Pull request by s-perron**
In previous changes, the option `--skip-validation` was disabled.  This
change is to reenable it.

### CODE REVIEW

File: include/spirv-tools/optimizer.hpp [hunk 1]
```diff
@@ -151,16 +151,22 @@ class Optimizer {
...
+  // occur when processing |original_binary| using any of the registered passes.
+  // In that case, no further passes are executed and the contents in
+  // |optimized_binary| may be invalid.
   //
   // It's allowed to alias |original_binary| to the start of |optimized_binary|.
   bool Run(const uint32_t* original_binary, size_t original_binary_size,
            std::vector<uint32_t>* optimized_binary) const;
-  bool Run(const uint32_t* original_binary, size_t original_binary_size,
+
+  // Same as above, except is passes |options| to the validator when trying to
```
Comment (dj2) [inline include/spirv-tools/optimizer.hpp]: Same as above, except passes

**Comment by johnkslang** [comment]
There is a problem here in that this, without an explicit request to do so, writes to stderr, which may not fit at all into a particular tool chain's use, especially programmatic use.

A tool should request that messages go to stderr, not have this happen by default.

**Comment by s-perron** [comment]
@johnkslang Are you sure your comment was for this PR?  It does not add any output stderr, as far as I can tell.

**Comment by johnkslang** [comment]
I am not sure, sorry.

What I saw was glslang's default use of SPIR-V tools getting double runs of the validator: one it requested and one it did not request, and the one it did not request was writing to stderr. My point is that should not be a default unrequested behavior; it should specifically requested, and it seemed directly tied to how this option is implemented in the API.

**Pull request merged**